onnxruntime/core/language_interop_ops/pyop/pyop.cc

```diff
@@ -93,32 +93,48 @@ void PyCustomKernel::Compute(OrtKernelContext* context) {
 int32_t PyCustomKernel::GetType(const OrtValue* input) const {
   int32_t numpy_type;
   ORT_ENFORCE(nullptr != input);
-  ORT_ENFORCE(const_cast<MLValue*>(input)->IsTensor(), "input must be a tensor");
-  auto data_type = const_cast<MLValue*>(input)->Get<Tensor>().DataType();
-  if (data_type == DataTypeImpl::GetType<bool>()) {
-    numpy_type = 0;
-  } else if (data_type == DataTypeImpl::GetType<int8_t>()) {
-    numpy_type = 1;
-  } else if (data_type == DataTypeImpl::GetType<uint8_t>()) {
-    numpy_type = 2;
-  } else if (data_type == DataTypeImpl::GetType<int16_t>()) {
-    numpy_type = 3;
-  } else if (data_type == DataTypeImpl::GetType<uint16_t>()) {
-    numpy_type = 4;
-  } else if (data_type == DataTypeImpl::GetType<int32_t>()) {
-    numpy_type = 5;
-  } else if (data_type == DataTypeImpl::GetType<uint32_t>()) {
-    numpy_type = 6;
-  } else if (data_type == DataTypeImpl::GetType<int64_t>()) {
-    numpy_type = 9;
-  } else if (data_type == DataTypeImpl::GetType<uint64_t>()) {
-    numpy_type = 10;
-  } else if (data_type == DataTypeImpl::GetType<float>()) {
-    numpy_type = 11;
-  } else if (data_type == DataTypeImpl::GetType<double>()) {
-    numpy_type = 12;
-  } else
-    ORT_ENFORCE(false, "Input type not supported");
+  ORT_ENFORCE(input->IsTensor(), "input must be a tensor");
+  auto data_type = input->Get<Tensor>().DataType()->AsPrimitiveDataType();
+  ORT_ENFORCE(data_type != nullptr, "Input type not supported");
+
+  namespace on = ONNX_NAMESPACE;
+  switch (data_type->GetDataType ()) {
+    case on::TensorProto_DataType_BOOL:
```

Comment:
Tensor::DataType()->AsPrimitiveDataType()->GetDataType() isn't overly user friendly. Can we add GetElementType() to Tensor to perform these steps and make usage simpler?

Reply:
Replaced with Tensor method where applicable.

---
In reply to: [342886651](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r342886651) [](ancestors = 342886651)